backend/src/schema/resolvers/registration.spec.js

```diff
@@ -118,194 +127,224 @@ describe('CreateInvitationCode', () => {
+          it('with a `createdAt` attribute', async () => {
+            await mutate({ mutation, variables })
+            let emailAddress = await neode.first('EmailAddress', { email: 'someuser@example.org' })
+            emailAddress = await emailAddress.toJson()
+            expect(emailAddress.createdAt).toBeTruthy()
+            expect(Date.parse(emailAddress.createdAt)).toEqual(expect.any(Number))
           })
 
-          it('throws unique violation error', async () => {
-            await expect(action()).rejects.toThrow('User account with this email already exists.')
+          it('with a cryptographic `nonce`', async () => {
+            await mutate({ mutation, variables })
+            let emailAddress = await neode.first('EmailAddress', { email: 'someuser@example.org' })
+            emailAddress = await emailAddress.toJson()
+            expect(emailAddress.nonce).toEqual(expect.any(String))
+          })
+
+          it('connects inviter through invitation code', async () => {
+            await mutate({ mutation, variables })
+            const result = await neode.cypher(
+              'MATCH(inviter:User)-[:GENERATED]->(:InvitationCode)-[:ACTIVATED]->(email:EmailAddress {email: {email}}) RETURN inviter',
+              { email: 'someuser@example.org' },
+            )
+            const inviter = neode.hydrateFirst(result, 'inviter', neode.model('User'))
+            await expect(inviter.toJson()).resolves.toEqual(
+              expect.objectContaining({ name: 'Inviter' }),
+            )
           })
-        })
 
-        describe('if the EmailAddress already exists but without user account', () => {
-          // shall we re-send the registration email?
-          it.todo('decide what to do')
+          describe('using the same InvitationCode twice', () => {
+            it('rejects because codes can be used only once', async () => {
+              await mutate({ mutation, variables })
+              variables = { ...variables, email: 'yetanotheremail@example.org' }
+              await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+                errors: [
+                  { message: 'UserInputError: Invitation code already used or does not exist.' },
+                ],
+              })
+            })
+          })
+
+          describe('if a user account with the given email already exists', () => {
+            beforeEach(async () => {
+              await factory.create('User', { email: 'someuser@example.org' })
+            })
+
+            it('throws unique violation error', async () => {
+              await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+                errors: [{ message: 'User account with this email already exists.' }],
+              })
+            })
+          })
+
+          describe('if the EmailAddress already exists but without user account', () => {
+            it.todo('shall we re-send the registration email?')
+          })
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-06T12:48:37Z" title="Friday, September 6th 2019, 2:48:37 pm +02:00">Sep 6, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -118,194 +127,225 @@ describe('CreateInvitationCode', () => {
 })
 
 describe('SignupByInvitation', () => {
-  const mutation = `mutation($email: String!, $token: String!) {
-    SignupByInvitation(email: $email, token: $token) { email }
-  }`
-
-  beforeEach(() => {
-    client = new GraphQLClient(host)
-    action = async () => {
-      return client.request(mutation, variables)
+  const mutation = gql`
+    mutation($email: String!, $token: String!) {
+      SignupByInvitation(email: $email, token: $token) {
+        email
+      }
     }
-  })
+  `
 
   describe('with valid email but invalid InvitationCode', () => {
     beforeEach(() => {
-      variables.email = 'any-email@example.org'
-      variables.token = 'wut?'
-    })
-
-    it('throws UserInputError', async () => {
-      await expect(action()).rejects.toThrow('Invitation code already used or does not exist.')
-    })
-  })
-
-  describe('with valid InvitationCode', () => {
-    beforeEach(async () => {
-      const inviterParams = {
-        name: 'Inviter',
-        email: 'inviter@example.org',
-        password: '1234',
+      variables = {
+        ...variables,
+        email: 'any-email@example.org',
+        token: 'wut?',
       }
-      const factory = Factory()
-      await factory.create('User', inviterParams)
-      const headersOfInviter = await login(inviterParams)
-      const anotherClient = new GraphQLClient(host, { headers: headersOfInviter })
-      const invitationMutation = `mutation { CreateInvitationCode { token } }`
-      const {
-        CreateInvitationCode: { token },
-      } = await anotherClient.request(invitationMutation)
-      variables.token = token
     })
 
-    describe('given an invalid email', () => {
-      beforeEach(() => {
-        variables.email = 'someuser'
-      })
-
-      it('throws `email is not a valid email`', async () => {
-        await expect(action()).rejects.toThrow('"email" must be a valid email')
-      })
-
-      it('creates no additional EmailAddress node', async done => {
-        try {
-          await action()
-        } catch (e) {
-          let emailAddresses = await instance.all('EmailAddress')
-          emailAddresses = await emailAddresses.toJson
-          expect(emailAddresses).toHaveLength(0)
-          done()
-        }
+    it('throws UserInputError', async () => {
+      await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+        errors: [{ message: 'UserInputError: Invitation code already used or does not exist.' }],
       })
     })
 
-    describe('given a valid email', () => {
-      beforeEach(() => {
-        variables.email = 'someUser@example.org'
+    describe('with valid InvitationCode', () => {
+      beforeEach(async () => {
+        const inviter = await factory.create('User', {
+          name: 'Inviter',
+          email: 'inviter@example.org',
+          password: '1234',
+        })
+        authenticatedUser = await inviter.toJson()
+        const invitationMutation = gql`
+          mutation {
+            CreateInvitationCode {
+              token
+            }
+          }
+        `
+        const {
+          data: {
+            CreateInvitationCode: { token },
+          },
+        } = await mutate({ mutation: invitationMutation })
+        authenticatedUser = null
+        variables = {
+          ...variables,
+          token,
+        }
       })
 
-      it('resolves', async () => {
-        await expect(action()).resolves.toEqual({
-          SignupByInvitation: { email: 'someuser@example.org' },
+      describe('given an invalid email', () => {
+        beforeEach(() => {
+          variables = { ...variables, email: 'someuser' }
         })
-      })
 
-      describe('creates a EmailAddress node', () => {
-        it('with a `createdAt` attribute', async () => {
-          await action()
-          let emailAddress = await instance.first('EmailAddress', { email: 'someuser@example.org' })
-          emailAddress = await emailAddress.toJson()
-          expect(emailAddress.createdAt).toBeTruthy()
-          expect(Date.parse(emailAddress.createdAt)).toEqual(expect.any(Number))
+        it('throws `email is not a valid email`', async () => {
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+            errors: [{ message: expect.stringContaining('"email" must be a valid email') }],
+          })
         })
 
-        it('with a cryptographic `nonce`', async () => {
-          await action()
-          let emailAddress = await instance.first('EmailAddress', { email: 'someuser@example.org' })
-          emailAddress = await emailAddress.toJson()
-          expect(emailAddress.nonce).toEqual(expect.any(String))
+        it('creates no additional EmailAddress node', async () => {
+          let emailAddresses = await neode.all('EmailAddress')
+          emailAddresses = await emailAddresses.toJson()
+          expect(emailAddresses).toHaveLength(1)
+          await mutate({ mutation, variables })
+          emailAddresses = await neode.all('EmailAddress')
+          emailAddresses = await emailAddresses.toJson()
+          expect(emailAddresses).toHaveLength(1)
         })
+      })
 
-        it('connects inviter through invitation code', async () => {
-          await action()
-          const result = await instance.cypher(
-            'MATCH(inviter:User)-[:GENERATED]->(:InvitationCode)-[:ACTIVATED]->(email:EmailAddress {email: {email}}) RETURN inviter',
-            { email: 'someuser@example.org' },
-          )
-          const inviter = instance.hydrateFirst(result, 'inviter', instance.model('User'))
-          await expect(inviter.toJson()).resolves.toEqual(
-            expect.objectContaining({ name: 'Inviter' }),
-          )
+      describe('given a valid email', () => {
+        beforeEach(() => {
+          variables = { ...variables, email: 'someUser@example.org' }
         })
 
-        describe('using the same InvitationCode twice', () => {
-          it('rejects because codes can be used only once', async done => {
-            await action()
-            try {
-              variables.email = 'yetanotheremail@example.org'
-              await action()
-            } catch (e) {
-              expect(e.message).toMatch(/Invitation code already used/)
-              done()
-            }
+        it('resolves', async () => {
+          await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+            data: { SignupByInvitation: { email: 'someuser@example.org' } },
           })
         })
 
-        describe('if a user account with the given email already exists', () => {
-          beforeEach(async () => {
-            await factory.create('User', { email: 'someuser@example.org' })
+        describe('creates a EmailAddress node', () => {
+          it('with a `createdAt` attribute', async () => {
+            await mutate({ mutation, variables })
+            let emailAddress = await neode.first('EmailAddress', { email: 'someuser@example.org' })
+            emailAddress = await emailAddress.toJson()
+            expect(emailAddress.createdAt).toBeTruthy()
+            expect(Date.parse(emailAddress.createdAt)).toEqual(expect.any(Number))
           })
 
-          it('throws unique violation error', async () => {
-            await expect(action()).rejects.toThrow('User account with this email already exists.')
+          it('with a cryptographic `nonce`', async () => {
+            await mutate({ mutation, variables })
+            let emailAddress = await neode.first('EmailAddress', { email: 'someuser@example.org' })
+            emailAddress = await emailAddress.toJson()
+            expect(emailAddress.nonce).toEqual(expect.any(String))
+          })
+
+          it('connects inviter through invitation code', async () => {
+            await mutate({ mutation, variables })
+            const result = await neode.cypher(
+              'MATCH(inviter:User)-[:GENERATED]->(:InvitationCode)-[:ACTIVATED]->(email:EmailAddress {email: {email}}) RETURN inviter',
+              { email: 'someuser@example.org' },
+            )
+            const inviter = neode.hydrateFirst(result, 'inviter', neode.model('User'))
+            await expect(inviter.toJson()).resolves.toEqual(
+              expect.objectContaining({ name: 'Inviter' }),
+            )
           })
-        })
 
-        describe('if the EmailAddress already exists but without user account', () => {
-          // shall we re-send the registration email?
-          it.todo('decide what to do')
+          describe('using the same InvitationCode twice', () => {
+            it('rejects because codes can be used only once', async () => {
+              await mutate({ mutation, variables })
+              variables = { ...variables, email: 'yetanotheremail@example.org' }
+              await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+                errors: [
+                  { message: 'UserInputError: Invitation code already used or does not exist.' },
+                ],
+              })
+            })
+          })
+
+          describe('if a user account with the given email already exists', () => {
+            beforeEach(async () => {
+              await factory.create('User', { email: 'someuser@example.org' })
+            })
+
+            it('throws unique violation error', async () => {
+              await expect(mutate({ mutation, variables })).resolves.toMatchObject({
+                errors: [{ message: 'User account with this email already exists.' }],
+              })
+            })
+          })
+
+          describe('if the EmailAddress already exists but without user account', () => {
+            // shall we re-send the registration email?
+            it.todo('decide what to do')
```


The comment should be in the todo.
Otherwise it's not visible, if the tests are running.